wcfsetup/install/files/lib/acp/form/StyleEditForm.class.php

```diff
@@ -139,19 +139,54 @@ protected function readStyleVariables() {
 		}
 		
 		if ($this->variables['pageLogo']) {
-			$file = new UploadFile($this->style->getAssetPath().$this->variables['pageLogo'], basename($this->variables['pageLogo']), true, true, true);
-			UploadHandler::getInstance()->registerFilesByField('pageLogo', [
-				$file,
-			]);
+			if (Url::is($this->variables['pageLogo'])) {
+				$filename = basename(Url::parse($this->variables['pageLogo'])['path']);
+				$this->getLogoFromUrl($this->variables['pageLogo'], $this->style->getAssetPath().$filename);
+				$this->variables['pageLogo'] = $filename;
+			}
+			if (file_exists($this->style->getAssetPath().$this->variables['pageLogo'])) {
+				$file = new UploadFile($this->style->getAssetPath().$this->variables['pageLogo'], basename($this->variables['pageLogo']), true, true, true);
+				UploadHandler::getInstance()->registerFilesByField('pageLogo', [
+					$file,
+				]);
+			}
+			else {
+				$this->variables['pageLogo'] = '';
+			}
 		}
 		if ($this->variables['pageLogoMobile']) {
-			$file = new UploadFile($this->style->getAssetPath().$this->variables['pageLogoMobile'], basename($this->variables['pageLogoMobile']), true, true, true);
-			UploadHandler::getInstance()->registerFilesByField('pageLogoMobile', [
-				$file,
-			]);
+			if (Url::is($this->variables['pageLogoMobile'])) {
+				$filename = basename(Url::parse($this->variables['pageLogoMobile'])['path']);
```

Comment:
You must not trust the remote extension (and by extension the remote filename). Please validate the path against a sane whitelist (gif, png, jpeg, jpg, svg) and ignore the logo otherwise.